src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -585,33 +866,47 @@ public void AssemblyLoading_MultipleVersions_ExactAndGreaterMatch()
                 var e = epsilon.CreateInstance("Epsilon.E")!;
                 e.GetType().GetMethod("Write")!.Invoke(e, new object[] { sb, "Test E" });
 
-                VerifyDependencyAssemblies(
-                    loader,
-                    testFixture.Delta2B.Path,
-                    testFixture.Epsilon.Path);
-
                 var actual = sb.ToString();
-                if (ExecutionConditionUtil.IsCoreClr)
+                if (ExecutionConditionUtil.IsCoreClr || loader is ShadowCopyAnalyzerAssemblyLoader)
                 {
+                    // This works in CoreClr because we have full control over assembly loading. It 
+                    // works in shadow copy because all the DLLs are put into different directories
+                    // so everything is a AppDomain.AssemblyResolve event and we get full control there.
+                    VerifyDependencyAssemblies(
+                        loader,
+                        testFixture.Delta2B.Path,
+                        testFixture.Epsilon.Path);
+
                     Assert.Equal(
     @"Delta.2B: Epsilon: Test E
 ",
                         actual);
                 }
                 else
                 {
+                    // See limitation 2
+                    Assert.Throws<InvalidOperationException>(() => loader.GetRealLoadPath(testFixture.Delta2.Path));
+
+                    // Fake the dependency so we can verify the rest of the load
+                    loader.AddDependencyLocation(testFixture.Delta2.Path);
```

Comment:
Why is this necessary? Shouldn't we use `textFixture.Delta2B.Path` below instead?